### PR TITLE
cli: add a --durability= option for the database storage

### PR DESCRIPTION
**Pull request by nixpanic**
For testing and development environments where only a single storage pod
is available, the "heketidbstorage" volume can now be created without
any redundancy. The new --durability=... option defaults to "replicate",
but also accepts "none".

Signed-off-by: Niels de Vos <ndevos@redhat.com>

### CODE REVIEW

**Comment by nixpanic** [comment]
Updated indention/alignment of an option, Travis reported a `go fmt` error.

**Comment by obnoxxx** [comment]
ok to test

File: client/cli/go/cmds/heketi_storage.go [hunk 1]
```diff
@@ -113,9 +118,17 @@ func createHeketiStorageVolume(c *client.Client) (*api.VolumeInfoResponse, error
...
 	req := &api.VolumeCreateRequest{}
 	req.Size = HeketiStorageVolumeSize
-	req.Durability.Type = api.DurabilityReplicate
-	req.Durability.Replicate.Replica = 3
 	req.Name = db.HeketiStorageVolumeName
+	req.Durability.Type = dt
+
+	switch dt {
+	case api.DurabilityReplicate:
+		req.Durability.Replicate.Replica = 3
```
Comment (obnoxxx) [inline client/cli/go/cmds/heketi_storage.go]: Do we want to hardcode replica 3 here? or if replica is specified also offer the possibility to specify the count?...

Comment (nixpanic) [reply client/cli/go/cmds/heketi_storage.go]: This is what there was before. If we want to change it to an other type (arbiter?), that would be a different patch/pr.

Comment (nixpanic) [reply client/cli/go/cmds/heketi_storage.go]: Ah, yes, I see what you mean. That makes sense.

Do you want that in this patch, or a follow-up one?

Comment (obnoxxx) [reply client/cli/go/cmds/heketi_storage.go]: This one would be good. Simple amend and force-push over, please.

File: client/cli/go/cmds/heketi_storage.go [hunk 2]
```diff
@@ -55,6 +57,15 @@ func init() {
...
 		"container image to run this job")
+	setupHeketiStorageCommand.Flags().StringVar(&heketiStorageDurability,
+		"durability",
+		"replicate",
+		"Durability of the database storage")
+	setupHeketiStorageCommand.Flags().IntVar(&heketiStorageReplicaCount,
+		"replica",
+		3,
+		"\n\tReplica value for durability type 'replicate'."+
+			"\n\tDefault is 3")
```
Comment (obnoxxx) [inline client/cli/go/cmds/heketi_storage.go]: Here the formatting (line break / indentation) is a little inconsistent with the --durability option.
I suggest being a little more verbose and more or less copying the description of
--durability from the "volume create" command (omitting the disperse type).

File: client/cli/go/cmds/heketi_storage.go [hunk 3]
```diff
@@ -113,9 +124,17 @@ func createHeketiStorageVolume(c *client.Client) (*api.VolumeInfoResponse, error
...
 	req.Name = db.HeketiStorageVolumeName
+	req.Durability.Type = dt
+
+	switch dt {
+	case api.DurabilityReplicate:
+		req.Durability.Replicate.Replica = replicaCount
+	case api.DurabilityDistributeOnly:
+		// no further options needed
+	default:
+		return nil, fmt.Errorf("Durability %s id not supported for heketi database storage", dt)
```
Comment (obnoxxx) [inline client/cli/go/cmds/heketi_storage.go]: Typo: id -> is

File: client/cli/go/cmds/heketi_storage.go (hunk 2 above)
Comment (nixpanic) [reply client/cli/go/cmds/heketi_storage.go]: Done!

File: client/cli/go/cmds/heketi_storage.go (hunk 3 above)
Comment (nixpanic) [reply client/cli/go/cmds/heketi_storage.go]: Done!

**Comment by obnoxxx** [comment]
The code looks good to me now.
But unclear what acutally fails in the centos-ci.
It does *not* seem to be the cpu virtualization issue with libvirt/kvm.

**Comment by obnoxxx** [comment]
failure unrelated according to @phlogistonjohn

**Pull request merged**